packages/dds/migration-shim/src/migrationShim.ts

```diff
@@ -78,6 +87,22 @@ export class MigrationShim extends TypedEventEmitter<IMigrationEvent> implements
 		throw new Error("Method not implemented.");
 	}
 
+	public get target(): LegacySharedTree | ISharedTree {
+		return this.oldTree;
+	}
+
+	public async load(services: IChannelServices): Promise<void> {
+		this._oldTree = (await this.oldFactory.load(
+			this.runtime,
+			this.id,
+			services,
+			this.oldFactory.attributes,
+		)) as LegacySharedTree;
+	}
+	public create(): void {
+		this._oldTree = this.oldFactory.create(this.runtime, this.id);
+	}
```

Comment:
Should we permit creating an old tree?  Or should the shim always create the new tree only?  Or can we assume that new documents will always use a different container code with an un-shimmed new shared tree exclusively?

No change requested here, but something to think about as we iterate and explore integration.

Reply:
Not sure, I think it will depend on how Whiteboard wishes to ship. I would prefer if they load a new document to have their createDataObject try to create a SharedTree rather than a Legacy Shared tree. For now, this allows Whiteboard to make that decision themselves.